stdlib/getopt.pyi

```diff
@@ -1,7 +1,9 @@
+from collections.abc import Iterable
+
 __all__ = ["GetoptError", "error", "getopt", "gnu_getopt"]
 
-def getopt(args: list[str], shortopts: str, longopts: list[str] = []) -> tuple[list[tuple[str, str]], list[str]]: ...
-def gnu_getopt(args: list[str], shortopts: str, longopts: list[str] = []) -> tuple[list[tuple[str, str]], list[str]]: ...
+def getopt(args: list[str], shortopts: str, longopts: Iterable[str] = []) -> tuple[list[tuple[str, str]], list[str]]: ...
+def gnu_getopt(args: list[str], shortopts: str, longopts: Iterable[str] = []) -> tuple[list[tuple[str, str]], list[str]]: ...
```

Comment:
In fact, it can also be a `str`:

https://github.com/python/cpython/blob/c806cd5af677c385470001efc68da38a32919196/Lib/getopt.py#L81-L84

```suggestion
def getopt(args: list[str], shortopts: str, longopts: Iterable[str] | str = []) -> tuple[list[tuple[str, str]], list[str]]: ...
def gnu_getopt(args: list[str], shortopts: str, longopts: Iterable[str] | str = []) -> tuple[list[tuple[str, str]], list[str]]: ...
```

Reply:
I noticed that as well, but saw that the commit described it as "a common mistake", so I wasn't sure if it's something we'd want to explicitly support/encourage. Since `str` is an `Iterable[str]`, code should type check the same either way, right?

Reply:
Since it's explicitly handled, I think we should include it. And while it's true that `Iterable[str]` also includes `str`, it's better to be explicit here for documentation purposes. Also, there have been discussions in the past to possibly distinguish between those cases, and this would make this annotation future proof.

Reply:
Makes sense, sounds good! Made the change